src/gmail.js

```diff
@@ -1176,8 +1177,14 @@ var Gmail = function(localJQuery) {
 
         if (threadData && api.check.data.is_action(threadData)) {
             const action = api.tools.get_action(threadData);
-
-            return action_map[action];
+            
+            //Check if label is applied to email / existing email is moved to an label
+            if(action.startsWith(apply_label) && api.check.data.is_first_type_action(threadData)) {
+                return action_map[apply_label];
+            } else {
+                return action_map[action];
+            }
+            
         } else {
```

Comment:
We now have an `if` nested inside an `if`. I really don't like stuff like that. That's how spaghetti-code starts 😄 

Could you make this into a three-way `if`, `else if`, `else` instead? If you start with the `null`-condition, that would end up looking much cleaner.